sqlparser/test/parser/update_test.dart

```diff
@@ -26,6 +26,51 @@ void main() {
     testAll(testCases);
   });
 
+  test('parses updates with column-name-list and subquery', () {
+    testStatement(
+        '''
+      UPDATE foo
+      SET (a, b) = (SELECT b, a FROM bar AS b WHERE b.id=foo.id);
+      ''',
+        UpdateStatement(table: TableReference('foo'), set: [
+          MultiColumnSetComponent(
+              columns: [Reference(columnName: 'a'), Reference(columnName: 'b')],
+              rowValue: SubQuery(
+                  select: SelectStatement(
+                      columns: [
+                    ExpressionResultColumn(
+                      expression: Reference(columnName: 'b'),
+                    ),
+                    ExpressionResultColumn(
+                      expression: Reference(columnName: 'a'),
+                    ),
+                  ],
+                      from: TableReference('bar', as: 'b'),
+                      where: BinaryExpression(
+                        Reference(entityName: 'b', columnName: 'id'),
+                        token(TokenType.equal),
+                        Reference(entityName: 'foo', columnName: 'id'),
+                      ))))
```

Comment:
Let's add a few commas between the parentheses here so that `dart format` makes this look a little more tree-like :)